src/evaluator/functions.rs

```diff
@@ -997,6 +998,31 @@ pub fn fn_reverse<'a>(
     Ok(result)
 }
 
+#[allow(clippy::mutable_key_type)]
```

Comment:
I _think_ this is happening because technically, the `Value` that we're hashing is mutable? [ref](https://rust-lang.github.io/rust-clippy/master/index.html#/mutable_key_type)

e.g. if we have an array `[1, 2]`, add it to the hash-set, and then change the array to `[1, 2, 3]`, the hash also changes